stripe/src/main/java/com/stripe/android/model/SourceOrder.kt

```diff
@@ -0,0 +1,109 @@
+package com.stripe.android.model
+
+/**
+ * Information about the items and shipping associated with the source.
+ * Required for transactional credit (for example Klarna) sources before you can charge it.
+ *
+ * [API reference](https://stripe.com/docs/api/sources/object#source_object-source_order)
+ */
+data class SourceOrder(
+    /**
+     * A positive integer in the smallest currency unit (that is, 100 cents for $1.00, or 1 for ¥1,
+     * Japanese Yen being a zero-decimal currency) representing the total amount for the order.
+     */
+    val amount: Int? = null,
+
+    /**
+     * Three-letter [ISO currency code](https://www.iso.org/iso-4217-currency-codes.html),
+     * in lowercase. Must be a [supported currency](https://stripe.com/docs/currencies).
+     */
+    val currency: String? = null,
+
+    /**
+     * The email address of the customer placing the order.
+     */
+    val email: String? = null,
+
+    /**
+     * List of items constituting the order.
+     */
+    val items: List<String> = emptyList(),
+
+    /**
+     * The shipping address for the order. Present if the order is for goods to be shipped.
+     */
+    val shipping: Shipping? = null
+) {
+    /**
+     * List of items constituting the order.
+     *
+     * [API reference](https://stripe.com/docs/api/sources/object#source_object-source_order-items)
+     */
+    data class Item(
```

Comment:
out of curiosity, why nest these data classes and not separate them out?

Reply:
@chander-stripe they're small and simple enough that this seems fine, and they also have generic names (e.g. "Item", "Shipping") that might collide with other APIs